crates/optimism/rpc/src/eth/ext.rs

```diff
@@ -0,0 +1,132 @@
+use super::{OpEthApiInner, OpNodeCore};
+use crate::{error::Op4337Error, OpEthApiError, SequencerClient};
+use alloy_consensus::BlockHeader;
+use alloy_eips::BlockNumberOrTag;
+use alloy_primitives::{Bytes, B256};
+use alloy_rpc_types_eth::erc4337::TransactionConditional;
+use jsonrpsee_core::RpcResult;
+use op_alloy_network::TransactionResponse;
+use op_alloy_rpc_types::Transaction;
+use reth_provider::{BlockReaderIdExt, StateProviderFactory};
+use reth_rpc_eth_api::{L2EthApiExtServer, RpcNodeCore};
+use reth_transaction_pool::{TransactionOrigin, TransactionPool};
+use std::sync::Arc;
+
+/// Maximum execution const for conditional transactions.
+const MAX_CONDITIONAL_EXECUTION_COST: u64 = 5000;
+
+/// OP-Reth `Eth` API extensions implementation.
+///
+/// Separate from [`super::OpEthApi`] to allow to enable it conditionally,
+#[derive(Clone)]
+#[allow(dead_code)]
+pub(crate) struct OpEthApiExt<N: OpNodeCore> {
+    /// Gateway to node's core components.
+    inner: Arc<OpEthApiInner<N>>,
+}
+
+impl<N> OpEthApiExt<N>
+where
+    N: OpNodeCore<Provider: BlockReaderIdExt + Clone + 'static>,
+{
+    /// Returns the configured sequencer client, if any.
+    pub(crate) fn sequencer_client(&self) -> Option<&SequencerClient> {
+        self.inner.sequencer_client()
+    }
+}
+
+impl<N> RpcNodeCore for OpEthApiExt<N>
```

Comment:
do we need this, I believe we can remove this impl

Reply:
yep, we only needed 2 methods from this, moved to the basic impl